modules/core/src/umatrix.cpp

```diff
@@ -1257,9 +1265,16 @@ void UMat::copyTo(OutputArray _dst) const
         return;
     }
 
-    if( empty() )
+    if( dims == 0 && empty() )
     {
         _dst.release();
+        void* obj = _dst.getObj();
+        if (_dst.isMat())
+            reinterpret_cast<Mat*>(obj)->flags = Mat::MAGIC_VAL | Mat::CONTINUOUS_FLAG | type();
+        else if (_dst.isUMat())
+            reinterpret_cast<UMat*>(obj)->flags = UMat::MAGIC_VAL | UMat::CONTINUOUS_FLAG | type();
+        else if (_dst.isGpuMat())
+            reinterpret_cast<cuda::GpuMat*>(obj)->flags = type();
```

Comment:
should it be a part of dst.create()?

Reply:
please, look more carefully. This is called _after_ release(), not after create(). I found that empty matrix is not copied correctly, type is replaced with 'CV_8U', which is wrong in general. This code fixes it.

Reply:
fixed